pkg/monitor/cluster/cluster.go

```diff
@@ -91,14 +91,9 @@ func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftClu
 		return nil, err
 	}
 
-	var hiveclientset client.Client
```

Comment:
I think just setting this to nil would have covered the edge case.

Reply:
@jaitaiwan yes you are right. This is explained in the RCA. 😃 

The problem is that pkg/monitor/cluster/cluster.go.NewMonitor has implicit dependencies ("k8s.io/client-go/kubernetes", "github.com/openshift/client-go/machine/clientset/versioned", etc.). If we used dependency injection and inversion we would be able to mock the dependencies and test NewMonitor() but as it is not easy to do, creating an aux function enables us to at least test that new function.